apps/store/src/pages/[[...slug]].tsx

```
@@ -84,78 +79,55 @@ export const getStaticProps: GetStaticProps<PageProps, StoryblokQueryParams> = a
 
   const apolloClient = initializeApollo({ locale })
 
-  const timerName = `Get static props for ${locale}/${slug} ${draftMode ? '(draft)' : ''}`
-  console.time(timerName)
-  const version = draftMode ? 'draft' : undefined
-  const [layoutWithMenuProps, breadcrumbs, trustpilot] = await Promise.all([
-    getLayoutWithMenuProps(context, apolloClient),
-    fetchBreadcrumbs(slug, { version, locale }),
-    fetchTrustpilotData(),
-  ]).catch((error) => {
-    throw new Error(`Failed to fetch data for ${slug}: ${error.message}`, { cause: error })
-  })
-
-  if (layoutWithMenuProps === null) return { notFound: true }
-
-  let story: PageStory | ProductStory
   try {
-    story = await getStoryBySlug<PageStory | ProductStory>(slug, { version, locale })
-  } catch (error) {
-    console.info(`Story with slug ${locale}/${slug} not found`)
-    console.debug(error)
-    return { notFound: true }
-  } finally {
-    console.timeEnd(timerName)
-  }
-
-  const props = {
-    ...layoutWithMenuProps,
-    [STORY_PROP_NAME]: story,
-    breadcrumbs,
-    trustpilot,
-    hideChat: story.content.hideChat ?? false,
-  }
-
-  if (isProductStory(story)) {
-    const priceTemplate = fetchPriceTemplate(story.content.priceFormTemplateId)
-    if (priceTemplate === undefined) {
-      throw new Error(`Unknown price template: ${story.content.priceFormTemplateId}`)
+    const props = await getStoryblokPageProps({ context, slug, locale, draftMode })
```

This `getStaticProps` function will fail typescript here because `getLayoutWithMenuProps` can return the props or `null`. So whenever we call it we do the checking:

```
if (layoutWithMenuProps === null) return { notFound: true }
```

However we just want to spread those props inside `getStoryblokPageProps` which makes it difficult to check if layout with menu props was retrieved sucessfully while calling `getStoryblokPageProps`. So I thought about changing the signature of `getLayoutWithMenuProps` to throw an Error when it fails instead of `null` (next PR on the stack). Let me know what you think,